### PR TITLE
Unblacklist Intel igpu in single GPU configs

### PR DESCRIPTION
**Pull request by pkviet**
### Description
The blacklist of Intel igpu was motivated by issues with dual gpu setups in laptops.
A few people with old Intel igpu (which unfortunately have not  been documented) had also reported
issues with hardware acceleration in single gpu setups. Intel igpus were therefore blacklisted.
This however prevents the hardware decoding of webm videos by Intel igpu in single gpu setups.
With this commit hardware decoding of webm will be possible for Intel igpu in single gpu setups
while people who had issues with hardware acceleration can still disable it.

### Motivation and Context
Re-enables hardware acceleration for Intel igpu in single gpu setups.
They are still blacklisted in dual gpu setups.
Fixes [https://github.com/obsproject/obs-studio/issues/2248](https://github.com/obsproject/obs-studio/issues/2248).

### How Has This Been Tested?
Tested on a laptop with core i5 8250u and HD620 iGPU + win 10 Home.

### Types of changes
- Bug fix (hardware decoding of webm by Intel igpu)
- Breaking change (might perhaps cause issues for very old Intel igpus; these users can fix their issue by disabling hardware acceleration, in the rare odds they resurface.) 

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.

### CODE REVIEW

File: obs-browser-plugin.cpp [hunk 1]
```diff
@@ -524,15 +524,19 @@ bool obs_module_load(void)
...
+				if (!!wstrstri(deviceId.c_str(), *device)) {
+					hwaccel = false;
+					blog(LOG_INFO, "[obs-browser]: "
+						       "Blacklisted device "
+						       "detected, "
+						       "disabling browser "
+						       "source hardware "
+						       "acceleration.");
+					break;
+				}
```
Comment (notr1ch) [inline obs-browser-plugin.cpp]: The condition should probably be moved outside of the loop as it isn't influenced by the loop variables.

Comment (pkviet) [reply obs-browser-plugin.cpp]: done, thanks R1ch

**Comment by energizerfellow** [comment]
Might want to add a checkbox at the bottom of Advanced Settings, that's enabled by default, about disabling Intel GPU hardware video decoding acceleration if multiple GPUs are detected. It would be nice to at least have that as an option for the support peeps.  Put is under a section at the bottom of Advanced Settings called "Quirks" or something like that.

It would be nice to see a similar checkbox for the NVENC blacklist, but I digress.

**Comment by notr1ch** [comment]
There is already an option to disable browser source hardware acceleration, adding a second one which only takes effect with multiple GPUs would just be confusing. I doubt there are many people frequently switching between single and multi GPU setups to necessitate this.

**Pull request merged**